app/views/dashboards/show.html.erb

```diff
@@ -26,7 +26,14 @@
       <button class="btn btn-default" ng-click="decreaseEndTime()" title="Decrease end time">
         <i class="icon-fb"></i>
       </button>
-      <input type="text" class="form-control" datetime-picker datetime="globalConfig.endTime" data-format="yyyy-MM-dd" placeholder="Until">
+      <span class="until_container">
+        <input type="text" class="form-control" datetime-picker datetime="globalConfig.endTime" data-format="yyyy-MM-dd" placeholder="Until">
+        <span class="until_clear global pointer"
```

Comment:
The "bonus points" seem quite easy though: simply add this to the span:

    ng-show="globalConfig.endTime"

(same in the widget_time_options_template.html)